scripts/cgm_adapter.py

```diff
@@ -316,25 +349,43 @@ def parse_csv(self) -> Mapping[str, Any]:
             for row in reader:
                 row_num += 1
                 project_uuid = row['project_uuid']
-                project_shortname = row['project_shortname']
-                file_name = row['file_name']
-                file_source = row['file_source']
+                project_shortname = row['project_shortname'].strip()
+                file_name = row['file_name'].strip()
+                file_source = row['file_source'].strip()
+                catalog = row.get('catalog')  # Optional column
+                uploaded = row['uploaded']  # 0 or 1
+                date_added = row['date_added']
+                date_imported = row['date_imported']
+                file = File(name=file_name,
+                            source=file_source,
+                            project_id=project_uuid,
+                            row_num=row_num)
+                require('.' in file.new_name,
+                        f'File {file.new_name!r} has an invalid name')
+                require(file.new_name not in file_names,
+                        f'File {file.new_name!r} is not unique')
+                file_names.add(file.new_name)
+                if not uploaded == '1':
+                    log.info(f'Skipping row {row_num} file {file_name!r},'
+                             f' uploaded = {uploaded}')
+                    continue
+                if date_added and date_imported and date_imported >= date_added:
+                    log.info(f'Skipping row {row_num} file {file_name!r},'
+                             f' {date_imported!r} >= {date_added!r}')
```

Comment:
Not sure if this logging statement can benefit from [CONTRIBUTING](https://github.com/DataBiosphere/azul/blob/develop/CONTRIBUTING.rst#110logging) -> "We don't use f'' strings or string concatenation when interpolating dynamic values into log messages:"